python/ray/serve/_private/deployment_state.py

```diff
@@ -1245,44 +1288,44 @@ def _stop_wrong_version_replicas(self, max_to_stop=math.inf) -> int:
         )
         replicas_stopped = False
         code_version_changes = 0
-        user_config_changes = 0
+        reconfigure_changes = 0
         for replica in replicas_to_update:
-            # If the code version is a mismatch, we stop the replica. A new one
-            # with the correct version will be started later as part of the
-            # normal scale-up process.
-            if replica.version.code_version != self._target_state.version.code_version:
+            # If code version or ray actor options change, stop the replica. A new one
+            # with the correct version will be started later as part of the normal
+            # scale-up process.
+            if (
+                replica.version.code_version != self._target_state.version.code_version
+                or replica.version.ray_actor_options_hash
+                != self._target_state.version.ray_actor_options_hash
```

Comment:
I think this logic should live in some method of the `version` instead of exposing it here, `version.requires_actor_restart(new_version)` ?